app/src/androidTest/java/com/example/vybe/AddEditVibeActivityTest.java

```diff
@@ -84,34 +92,36 @@ public void SocialSpinnerTest(){
     }
 
     @Test
-    public void stringReasonValidTest(){
+    public void StringReason_EnterText_Valid(){
+        EditText editText = (EditText) solo.getView("reason_edit_text");
         //Click on Empty Text View
-        solo.clickOnView(solo.getView("reason_edit_text"));
+        solo.clickOnView(editText);
         //Enter the text: 'asdfg'
-        solo.clearEditText((android.widget.EditText) solo.getView("reason_edit_text"));
-        solo.enterText((android.widget.EditText) solo.getView("reason_edit_text"), "asdfg");
+        solo.clearEditText(editText);
+        solo.enterText(editText, "asdfg");
         solo.searchEditText("asdfg");
         // Validate the text on the TextView
         assertEquals("Text should be the field value", "asdfg",
-                ((EditText) solo.getView("reason_edit_text")).getText().toString());
+                (editText.getText().toString()));
 
     }
 
     @Test
-    public void stringReasonErrorTest(){
+    public void StringReason_EnterText_Invalid(){
         // test if more than 3 words is provided
-        solo.clickOnView(solo.getView("reason_edit_text"));
+        EditText editText = (EditText) solo.getView("reason_edit_text");
+        solo.clickOnView(editText);
         //Enter the text: 'as df ga sd'
-        solo.clearEditText((android.widget.EditText) solo.getView("reason_edit_text"));
-        solo.enterText((android.widget.EditText) solo.getView("reason_edit_text"), "as df ga sd");
+        solo.clearEditText(editText);
+        solo.enterText(editText, "as df ga sd");
         solo.clickOnButton("Save");
 
         assertEquals("Expecting an error", "Max 3 words allowed",
-                ((EditText) solo.getView("reason_edit_text")).getError().toString());
+                (editText.getError().toString()));
     }
 
     @Test
-    public void datePickerTest(){
+    public void DatePicker_SelectDate_Pass(){
```

Comment:
Just make note we will need to delete this test because we are removing the functionality for being able to select a date/time other than the current date/time.

Reply:
Comment has been added